### DISCUSSION TITLE
编译报错

### DISCUSSION THREAD

**Discussion by TTsXDU** [wh201906/SerialTest#52] [Q&A]
我用QtCreator5.13编译，出现了这样的报错问题，请问是什么原因呢，有什么解决办法呢？
![{925`IG6_W76{R_50M~24}O](https://github.com/wh201906/SerialTest/assets/121789556/09d6a9ed-1981-429c-bf5a-8256ef68a7b5)

**Comment by wh201906** [comment]
你需要下载QCustomPlot的源码，并将qcustomplot.cpp放到src文件夹下，再编译
下载链接可参考README当中的Linux编译步骤

**Reply by TTsXDU** [reply]
好的，我去试试

**Comment by TTsXDU** [comment]
博主好，我现在编译成功了，但上位机的蓝牙模块无法正常实现功能，请问是什么原因呢？






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "wh201906/SerialTest"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年2月26日(星期一) 下午3:35
***@***.***&gt;;
抄送:&nbsp;"Sire ***@***.******@***.***&gt;;
主题:&nbsp;Re: [wh201906/SerialTest] 编译报错 (Discussion #52)





 
你需要下载QCustomPlot的源码，并将qcustomplot.cpp放到src文件夹下，再编译
 下载链接可参考README当中的Linux编译步骤

**Reply by wh201906** [reply]
你使用的是什么编译器呢？之前有人反馈说MSVC编译出来的蓝牙不可用，需要使用MingW
#20 #37

**Comment by TTsXDU** [comment]
我使用的就是MinGW 64-bit的编译器






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "wh201906/SerialTest"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年2月26日(星期一) 下午4:15
***@***.***&gt;;
抄送:&nbsp;"Sire ***@***.******@***.***&gt;;
主题:&nbsp;Re: [wh201906/SerialTest] 编译报错 (Discussion #52)





 
你使用的是什么编译器呢？之前有人反馈说MSVC编译出来的蓝牙不可用，需要使用MingW
 #20 #37

**Reply by wh201906** [reply]
你说的无法正常实现功能具体是指什么情况呢？

**Reply by TTsXDU** [reply]
就是这里画红圈的地方，在编译之后是灰色的，有一行提示是qt.bluetooth: Dummy backend running. Qt Bluetooth module is non-functional.
![2024-02-26 160829](https://github.com/wh201906/SerialTest/assets/121789556/e84aa7ec-7647-4c9b-bdb5-191ca88a473f)

![E(H1FB}%ZBPBZ~XY%{8HODN](https://github.com/wh201906/SerialTest/assets/121789556/36d50f10-336f-4371-b0e8-f322a2445851)

**Reply by TTsXDU** [reply]
请问博主知道这是什么原因吗

**Reply by wh201906** [reply]
奇怪了，你用GitHub Releases里面编译好的版本会遇到这种情况吗？

**Reply by TTsXDU** [reply]
用您编译并打包好的上位机时，各个功能都正常。我是想自己在QT运行一下SerialTest.Pro这个工程文件，想了解的透彻一些，然后就出现这个蓝牙模块在编译后是灰色的点不了的情况。我看网上有人说是QTCreator版本不对，还有人说是编译器MSCV和MinGW都要下，我也不清楚是什么原因。您还有什么别的看法分析分析吗？
![2024-02-26 160829](https://github.com/wh201906/SerialTest/assets/121789556/760f72df-9fda-4197-9dcd-5b905d02b673)
![E(H1FB}%ZBPBZ~XY%{8HODN](https://github.com/wh201906/SerialTest/assets/121789556/6c9a3c10-ecdd-4a06-82d7-df2f80b7dca4)

**Reply by wh201906** [reply]
要不你把编译好的exe文件发给我我这边试试？

**Reply by TTsXDU** [reply]
[上位机.zip](https://github.com/wh201906/SerialTest/files/14404640/default.zip)
这是我自己运行后生成的上位机文件。
我刚才点这里面编译好的.exe文件，发现它提示这样的错误：
![image](https://github.com/wh201906/SerialTest/assets/121789556/ff4222e8-468a-4885-aba2-d69bbf369b40)

是不是因为缺少这个.dll文件导致错误

**Reply by wh201906** [reply]
> 是不是因为缺少这个.dll文件导致错误

不是。你的exe文件从Qt Creator当中启动的时候会自动加载Qt安装目录下的dll，所以能正常打开。但是你直接启动exe的时候没有这个效果

你可以生成release版本的吗？我需要用release版本的exe文件测试一下。Qt Creator左下角可以切换生成exe的类型
![图片](https://github.com/wh201906/SerialTest/assets/62299611/c3dad011-040d-4063-998e-1b5982202a66)

**Reply by TTsXDU** [reply]
这是我生成的Release版本的压缩包。
[上位机.zip](https://github.com/wh201906/SerialTest/files/14405087/default.zip)

**Reply by wh201906** [reply]
我这边用Qt5.15.2的dll配合你编译出来的exe可以正常使用，我猜测是Qt版本问题。
![图片](https://github.com/wh201906/SerialTest/assets/62299611/0aaf3283-6e0c-44f8-b68d-0f1226e81506)

测试下来Qt5.15.2的DLL和Qt5.13有一定的兼容性，你可以把这个压缩包里面的dll解压到SerialTest.exe所在的目录里面，然后运行，应该也能解决问题
[Qt5Bluetooth.zip](https://github.com/wh201906/SerialTest/files/14405163/Qt5Bluetooth.zip)

**Comment by TTsXDU** [comment]
好的好的，我去下个5.15.2版本试试，如有问题再向您请教。




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "wh201906/SerialTest"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年2月26日(星期一) 晚上9:18
***@***.***&gt;;
抄送:&nbsp;"Sire ***@***.******@***.***&gt;;
主题:&nbsp;Re: [wh201906/SerialTest] 编译报错 (Discussion #52)





 
我这边用Qt5.15.2的DLL配合你编译出来的exe可以正常使用，我猜测是Qt版本问题。

**Reply by wh201906** [reply]
刚才消息没编辑完就发送了。你可以先试一下我在上一帖里面提供的替代方案

**Reply by TTsXDU** [reply]
博主，QT5.15.2有安装包吗，我发现网上现在下不了5.15.2了，都是QT6 QAQ

**Reply by wh201906** [reply]
5.15.2只有在线安装版本。
刚才我提供的替代方案能用吗？

**Reply by TTsXDU** [reply]
博主你好，我试了把需要的dll文件解压到serialtest.exe文件中，点击运行会出现这样的错误：
![VQPAA3_0V6LO36 LWJA@K2L](https://github.com/wh201906/SerialTest/assets/121789556/07e47a40-9740-47f4-995d-baa3cabaece0)

**Reply by TTsXDU** [reply]
我可能还是需要下一个QT5.15，主要是通过在线安装模式最低版本都是QT6没有QT5.15。QAQ

**Reply by TTsXDU** [reply]
博主你好，我刚才下了QT5.14，编译后蓝牙模块能用了，估计就是QT版本的问题。麻烦博主了。

**Comment by TTsXDU** [comment]
好的，明白




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "wh201906/SerialTest"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年2月26日(星期一) 晚上9:22
***@***.***&gt;;
抄送:&nbsp;"Sire ***@***.******@***.***&gt;;
主题:&nbsp;Re: [wh201906/SerialTest] 编译报错 (Discussion #52)





 
刚才消息没编辑完就发送了。你可以先试一下我在上一帖里面提供的替代方案
